### PR TITLE
Appease the gods of code style

### PR DESCRIPTION
**Pull request by darrenleeweber**
revise code to appease the complaints of rubocop and/or hound.ci for
files related to changes in user permissions.

Resolved conflicts in:
	app/controllers/items_controller.rb
	app/controllers/registration_controller.rb

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -661,4 +749,23 @@ def enforce_versioning
     render status: :forbidden, text: 'Object cannot be modified in its current state.'
     false
   end
+
+  # ---
+  # Dor::Workflow utils
+
+  def dor_accession_error?(object, task)
```
Comment (mejackreed) [inline app/controllers/items_controller.rb]: Some similar type stuff over here https://github.com/sul-dlss/argo/blob/develop/app/controllers/workflow_service_controller.rb#L42-L79 . Not sure if that can be combined, but just wanted to point it out.

Comment (mejackreed) [inline app/controllers/items_controller.rb]: Ahah! Saw your other work now.

Comment (darrenleeweber) [inline app/controllers/items_controller.rb]: Watch those typos: https://github.com/sul-dlss/argo/blob/develop/app/controllers/workflow_service_controller.rb#L74

Comment (darrenleeweber) [inline app/controllers/items_controller.rb]: Also slightly modified in the APO PR at https://github.com/sul-dlss/argo/blob/issue76-apo-nice/app/controllers/workflow_service_controller.rb

Comment (mejackreed) [inline app/controllers/items_controller.rb]: Isn't that the european way to spell it? :smiling_imp:

**Comment by drh-stanford** [comment]
It looks like `develop` is not rubocop-clean overall, but if we're overhauling the style for these files I prefer to have them pass rubocop -- I'm still seeing errors in the controllers in the PR:

```
[drh@macbookpro 15:32 argo (code-styles-from-apo-pr)] % rubocop -f simple app/controllers/apo_controller.rb app/controllers/items_controller.rb app/controllers/registration_controller.rb app/controllers/workflow_service_controller.rb 
Warning: unrecognized cop Style/SingleSpaceBeforeFirstArg found in /Users/drh/workspace/argo/.rubocop_todo.yml
Warning: unrecognized cop Style/SingleSpaceBeforeFirstArg found in /Users/drh/workspace/argo/.rubocop.yml
== app/controllers/apo_controller.rb ==
C: 21: 32: Freeze mutable objects assigned to constants.
C: 38:  5: Use the return of the conditional for variable assignment and comparison.
C:189: 53: Convert if nested inside else to elsif.
C:197:  5: Use the return of the conditional for variable assignment and comparison.
C:202:  5: Use the return of the conditional for variable assignment and comparison.
C:333: 68: Add parentheses to nested method call current_user.roles params[:id].
== app/controllers/items_controller.rb ==
C:  1:  1: Class has too many lines. [632/528]
C: 10: 14: Avoid comma after the last item of an array.
C: 24: 23: Avoid comma after the last item of an array.
C: 28: 24: Avoid comma after the last item of an array.
C: 36: 19: Avoid comma after the last item of an array.
C: 46: 22: Avoid comma after the last item of an array.
C: 51: 20: Avoid comma after the last item of an array.
C:101:  9: Use the return of the conditional for variable assignment and comparison.
W:207: 22: Unused block argument - format. You can omit the argument if you don't care about it.
C:208:  9: Use a guard clause instead of wrapping the code inside a conditional expression.
C:735: 73: Add parentheses to nested method call current_user.roles @apo.
C:741: 71: Add parentheses to nested method call current_user.roles @apo.
== app/controllers/registration_controller.rb ==
C:106:181: Line is too long. [193/180]

4 files inspected, 19 offenses detected
```

I'm ok with disabling the checks in the code comments, like so https://github.com/bbatsov/rubocop#disabling-cops-within-source-code, or by updating our rubocop_todo.yml.

**Comment by mejackreed** [comment]
Looks good overall. I created a tech debt ticket about combining the similar WFS calls #544

File: app/controllers/status_controller.rb [hunk 2]
```diff
@@ -17,21 +17,29 @@ def log
...
     else
-      render :status => 500, :text => "Nothing indexed recently. Even after saving '#{params[:test_obj]}'."
-      return
+      msg = "Nothing indexed recently. Even after saving '#{params[:test_obj]}'."
+      render status: 500, text: msg and return
     end
   rescue ActiveFedora::ObjectNotFoundError
-    render :status => 404, :text => "No object '#{params[:test_obj]}' found"
+    msg = "No object '#{params[:test_obj]}' found"
+    render status: 404, text: msg and return
```
Comment (darrenleeweber) [inline app/controllers/status_controller.rb]: rubocop complains about this common rails pattern also, what are we to do?

File: app/controllers/registration_controller.rb [hunk 3]
```diff
@@ -103,8 +103,19 @@ def rights_list
...
+      '*:*',
+      {
+        rows: 0,
+        facets: facet_fields,
+        :'facet.prefix' => params[:term].titlecase,
+        :'facet.mincount' => 1,
+        :'facet.limit' => 15
+      }
+    )
+    result = response.facets.find { |f| f.name == facet_field }.items.map(&:value).sort
```
Comment (darrenleeweber) [inline app/controllers/registration_controller.rb]: I've rare used either of these, usually `select` is my goto.

Comment (darrenleeweber) [inline app/controllers/registration_controller.rb]: See also https://github.com/bbatsov/rubocop/issues/1334, where they propose to disable this.

**Comment by darrenleeweber** [comment]
The comments from @drh-stanford are now incorporated.  I think this is blocking further work on the APO blocker.

**Pull request merged**